### DISCUSSION TITLE
How ignore model change

### DISCUSSION THREAD

**Discussion by leandrogmuller** [sqlalchemy/sqlalchemy#6308] [Usage Questions]
Hello.

How can I ignore changes on model?
Example.

engine = create_engine("postgresql://user:pass@localhost:5432/datadev",
                       poolclass=NullPool)
session_factory = sessionmaker(bind=engine)
session = session_factory()
lang03 = session.query(Lang03).filter(Lang03.pkge == 'kernel').first()
lang03.text = 'd'    # I dont't whant to change on database.
session.commit()
session.close()

**Comment by CaselIT** [comment]
hi,

you can use expunge to remove an object from a session. https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=merge#sqlalchemy.orm.Session.expunge

that way its changes will no longer tracked by the session

**Reply by zzzeek** [reply]
also note you can set an attribute on an object without making any "history" using attributes.set_committed_value: https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=set_committed_value#sqlalchemy.orm.attributes.set_committed_value .  this is usually used when you're setting an attribute to be equivlaent to what would have been loaded from the database.
